packages/hadron-app-registry/README.md

```diff
@@ -1,36 +1,165 @@
-# hadron-app-registry [![][npm_img]][npm_url]
+# hadron-app-registry
 
-> Hadron App Registry
+## Concepts
 
-## Installation
+> [!IMPORTANT]
+> Use a plugin instead of a regular component for self-contained parts of
+> the application which may need to keep state beyond the raw React
+> component's lifetime.
 
+Compass uses a special form of React components referred to as **Plugins**,
+distinguished from regular components in that they:
+
+- Maintain state that can persist beyond their React component's lifetime
+- Use a custom dependency injection mechanism for consuming dependencies that
+  are used when initializing the plugin state
+
+These dependencies are generally referred to as **Services**. For example,
+logging or telemetry in Compass are generally consumed by plugins as a
+service.
+
+Compass uses a concept of scopes called **App Registries**. Currently, there
+are two levels of nesting:
+
+- The global app registry, global in the sense of being a per-browser-window
+  singleton
+- A local app registry, where "local" currently means scoped to a single tab
+  (not a technical restriction, just a convention).
+
+The lifetime of a plugin's state is not tied to the lifetime of their React
+component, but rather to the lifetime of the local app registry (or the global
+one, if there is none).
+
+> [!TIP]
+> Most plugins in Compass use a **Redux store** to keep track of their state.
+> Some legacy plugins may still use **Reflux** or a similar mechanism where the
+> state is provided to the top-level component of the plugin as a plain object.
+
+Other than tracking plugin lifetimes, app registries provide a communication
+channel between plugins by being event emitters.
+If possible, new code should avoid this communication channel, as it is untyped
+and the specific method of passing messages should be an implementation detail.
+Alternative methods of passing messages between plugins include using React
+contexts to provide an API that can be used by nested plugins, or if that is
+not possible, your plugin can expose methods that other plugins can then access.
```

Comment:
What do you feel about including WorkspaceService as an example here?

Reply:
I've added a short note in 36a52c08c0871196358653d525b19b140c52a372 but I don't really want to optimize for this (hopefully) rather uncommon case here :)